spec/requests/trains_request_spec.rb

```diff
@@ -1,24 +1,77 @@
 require 'rails_helper'
 
 describe "Trains" do
-  before do
-    expect_any_instance_of(TrainApi).to receive(:index).and_return(
-      [{
-        "id" => 1,
-        "name" => "name",
-        "train_model_name" => "train_model_name",
-        "number_of_cars" => 2,
-        "max_weight_capacity" => 100,
-        "active" => true
-      }]
-    )
+  let(:valid_attributes) do
+    {
+      "id" => 1,
+      "name" => "name",
+      "train_model_name" => "train_model_name",
+      "number_of_cars" => 2,
+      "max_weight_capacity" => 100,
+      "active" => true
+    }
   end
```

Comment:
if this is supposed to represent a train how about we call this variable a `train` since its a bit odd seeing `valid_attributes` be used as a signifier for getting back a train